packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js

```diff
@@ -397,8 +397,14 @@ export default {
             });
           }
 
-          // Ignore references to the function itself as it's not defined yet.
           const def = reference.resolved.defs[0];
+
+          // Ignore if def is undefined - e.g. `arguments` binding.
+          if (def === undefined) {
```

Comment:
Let's make this sloppy `== null` and then remove `!= null` from next comparison.

Reply:
Good shout! I've updated the PR.